src/py_pecg/filtering_module.py

```diff
@@ -0,0 +1,564 @@
+    """
+    Resample data at a higher rate using lowpass interpolation.
+    Resamples the sequence in vector X at R times the original sample rate.
+    The resulting resampled vector Y is R times longer, LENGTH(Y) = R*LENGTH(X).
+    A symmetric filter, B, allows the original data to pass through
+    unchanged and interpolates between so that the mean square error
+    between them and their ideal values is minimized.
+
+    Args:
+        idata (np.ndarray): The target signal designated for interpolation.
+        r (int): The interpolation factor of the resulting resampled vector.
+        n (int): Half the number of original sample values used to perform the interpolation.
+                 For best results, use N no larger than 10. The length of B is 2*N*R+1.
+        cutoff (float): Cutoff Frequency. The signal is assumed to be band
+                        limited with cutoff frequency 0 < CUTOFF <= 1.0.
+
+    Raises:
+        TypeError: Type validaiton for r,n and cutoff.
+        ValueError: Value validation for cutoff.
+
+    Returns:
+        b (np.ndarray): returns the coefficients of the interpolation filter B.
+        odata (np.ndarray): returns the interpolated version of the target signal.
+
+    Notes:
+        i: Filter a fabricated section of data first (match initial values and first
+        derivatives by rotating the first data points by 180 degrees) to get
+        guess of good initial conditions. Filter length is 2*r*n+1 so need that
+        many points; can't duplicate first point or guarantee a zero slope at
+        beginning of sequence.
+        ii: Make sure right hand points of data have been correctly interpolated and
+        get rid of transients by again matching end values and derivatives of the
+        original data.
+
+    Credits:
+        Copyright 1988-2022 The MathWorks, Inc.
+        "Programs for Digital Signal Processing", IEEE Press
+        John Wiley & Sons, 1979, Chap. 8.1.
+    """
+
+    def designInterpFilt(r: int, n: int, alpha: float) -> np.ndarray:
+        """
+        This function designs a linear-phase FIR filter of type I for interpolation.
+
+        Args:
+            r (int): The interpolation factor of the resulting resampled vector.
+            n (int): Half the number of original sample values used to perform the interpolation.
+            alpha (float): Variation of the cutoff frequency specification.
+
+        Returns:
+            b (TYPE): The resultant interpolation filter coefficients.
+        """
+        # filter specification (frequency and magnitude)
+        if alpha == 1:
+            M = np.array([r, r, 0, 0])
+            F = np.array([0, 1 / (2 * r), 1 / (2 * r), 0.5])
+        else:
+            Nband = int((np.floor(0.5 * r)))
+            M = np.concatenate(
+                (np.array([r, r]), np.zeros((1, 2 * Nband)).flatten())
+            )
+            a2r = alpha / 2 / r
+            F = np.zeros((1, 2 * Nband + 2)).flatten()
```

Comment:
No need to create as a 2D and flatten, just create as 1D:
```suggestion
            F = np.zeros(2 * Nband + 2)
```